lib/libutee/arch/arm/ftrace/ftrace.c

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <setjmp.h>
+#include <user_ta_header.h>
+#include <utee_syscalls.h>
+#include "ftrace.h"
+
+static const char hex_str[] = "0123456789abcdef";
+static uint32_t ret_idx_setjmp;
```

Comment:
There can be multiple calls to `setjmp()` and any of the saved `jmp_buf` may be passed to `longjmp()`. Isn't there some other way to know how many stack frames you're unrolling?

Reply:
Ah, I see. In this case it seems I can use `jmp_buf` itself to save the `ret_idx`. It seems `jmp_buf +104` address isn't used, so will use it to save `ret_idx`.